### PR TITLE
Ada 42 ada 78 permissions sign out

### PR DESCRIPTION
**Pull request by jgeorge37**
(Reminder! this is a dev PR so only review for code quality/logic, no need to check out locally :) )

This covers basically everything relating to signing in, account types, logging out, etc. 

I moved the NavBar into the main app file since it can and should appear everywhere, and the user can log out from there. Logging out will remove their info from the local storage, show a notification confirming successful log out, and show a sign in button again. If they are on a page that requires them to be signed in to view, they will also no longer be able to see the contents of the page and get a generic "unauthorized" message.

For the permissions, I'm using a loading wheel while we check if the user is allowed to see the page and they will then receive an an unauthorized message if they are blocked. The user will also not be able to see options that are not allowed for them in the NavBar, so they are unlikely to try to go to those in the first place.

### CODE REVIEW

**Comment by jgeorge37** [comment]
not sure why it claims theres a file change from Sara, I did check that that file already exists in Dev so it most likely shouldn't cause an issue

File: acmw-db-app/pages/_app.js [hunk 1]
```diff
@@ -1,13 +1,62 @@
 import '../styles/globals.css';
 import Head from 'next/head';
+import NavBar from '../components/NavBar';
+import React, { useEffect, useState } from 'react';
+
+const loadingWheel = (
+  <div className="lds-ring"><div></div><div></div><div></div><div></div></div>
```
Comment (SnobbyDragon) [inline acmw-db-app/pages/_app.js:7]: can you indent this for clarity
this is so minor I'm going to approve anyways bc everything else looks good

**Review by SnobbyDragon — APPROVED** [review]

File: acmw-db-app/pages/_app.js (hunk 1 above)
Comment (jgeorge37) [reply acmw-db-app/pages/_app.js:7]: bruh its literally just a bunch of empty divs to make a circle that i stole from online do i have to

Comment (SnobbyDragon) [reply acmw-db-app/pages/_app.js:7]: okay fine but it made me scared

**Pull request merged**